### PR TITLE
Add optional bonus modules (closes #104)

### LINKED ISSUE #104: Allow for entire modules to be marked as "bonus"
E.g. "Dev-ops".

Closes #18

### PR DESCRIPTION
**Pull request by rontu85**
This PR closes #104 and adds the ability to separate modules in standard and bonus modules. Test coverage could be extended if required.

### CODE REVIEW

File: tests/Feature/ModulesTest.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        ]);
+
+        $bonusModule = factory(Module::class)->create([
+            'name' => 'Bonus module',
+            'is_bonus' => 1,
+        ]);
+
+        $response = $this->get("/en/modules");
+
+        $this->assertTrue(count($this->getResponseData($response, 'bonusModules')) == 1);
```
Comment (mattstauffer) [inline tests/Feature/ModulesTest.php]: I think you can replace these `getResponseData` assertions with `assertViewHas`: https://laravel.com/docs/6.x/http-tests#assert-view-has

File: tests/Feature/ModulesTest.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class ModulesTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /** @test */
+    function modules_can_be_seen()
```
Comment (mattstauffer) [inline tests/Feature/ModulesTest.php]:
```suggestion
    function module_index_page_lists_modules()
```

File: resources/views/modules/index.blade.php [hunk 3]
```diff
@@ -27,6 +27,21 @@
                 </li>
                 @endforeach
             </ul>
+
+            @if($bonusModules->count() > 0)
```
Comment (mattstauffer) [inline resources/views/modules/index.blade.php]:
```suggestion
            @if ($bonusModules->count() > 0)
```

File: tests/Feature/ModulesTest.php (hunk 1 above)
Comment (rontu85) [reply tests/Feature/ModulesTest.php]: A closure does the trick. I'll update it and send you a message here when it's done.

Comment (rontu85) [reply tests/Feature/ModulesTest.php]: [Matt](@mattstauffer), I've updated the assertions. Is this what you had in mind?

Comment (mattstauffer) [reply tests/Feature/ModulesTest.php]: @rontu85 hm, i'd be curious to debug that `getKey` error. I'll take a look as soon as i can get a chance!

Comment (rontu85) [reply tests/Feature/ModulesTest.php]: @mattstauffer no need to debug this. It just tries to compare an eloquent model with a collection which does not work. The method getKey should come from the model. As stated above a closure was necessary to solve this.

**Comment by mattstauffer** [comment]
Thanks @rontu85!

**Pull request merged**